lib/engine/game/base.rb

```diff
@@ -1121,8 +1122,19 @@ def value_for_sellable(player, corporation)
         max_bundle&.price || 0
       end
 
+      def president_sales_to_market?(corporation)
+        case self.class::PRESIDENT_SALES_TO_MARKET
+        when true
+          true
+        when ::Set
+          self.class::PRESIDENT_SALES_TO_MARKET.include?(corporation.id)
+        else
+          false
+        end
+      end
+
       def value_for_dumpable(player, corporation)
-        return value_for_sellable(player, corporation) if self.class::PRESIDENT_SALES_TO_MARKET
+        return value_for_sellable(player, corporation) if president_sales_to_market?(corporation)
```

Comment:
If there is still a problem with the `when ::Set` statement, then this call is going to affect games other than 18RoyalGeorge.

Reply:
Moved all code on this PR into 18RoyalGorge's game class only.